### PR TITLE
juan/W-12212031

### PR DESCRIPTION
**Pull request by JuanStenghele-sf**
### What does this PR do?

Adds the cache for pipeline deploy and calls it if the async flag is used.

### What issues does this PR fix or reference?

[W-12212031](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001HNc7qYAD/view)

### Functionality Before

The cache for pipeline deploy did not exist.

### Functionality After

Cache created for pipeline deploy with add(), unset(), update() and resolveLatest(). It it used when the --async flag is set.

### CODE REVIEW

File: messages/cache.md
Comment (JuanStenghele-sf) [inline messages/cache.md:1]: This needs Emily's approval

**Review by EstebanRomero84 — APPROVED** [review]
Nice work Juan!
Just let's wait for Emily's approval

File: src/common/deployPipelineCache.ts
Comment (molson-sfdc) [inline src/common/deployPipelineCache.ts]:
```suggestion
export type AsyncOperationData = Record<string, never>;
```

This might be a bit more readable.  Just a suggestion.

Comment (JuanStenghele-sf) [reply src/common/deployPipelineCache.ts]: Ok!

**Pull request merged**
